### PR TITLE
Dispatch the auxiliary frobenius_norm and nuclear_norm to better implementations and deprecate them

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #81763

These functions will be legacy functions. We deprecate them, but we also
take this chance to dispatch to a more efficient and consistent implementation.
Doing so should help writing a conversion rule for these to be able to
remove them once and for all

cc @jianyuh @nikitaved @pearu @mruberry @walterddr @IvanYashchuk @xwang233 @Lezcano @ezyang @ngimel @peterbell10

Differential Revision: [D42354776](https://our.internmc.facebook.com/intern/diff/D42354776)

### CODE REVIEW

**Comment by Lezcano** [comment]
About BC compatibility. We run the same test as described in https://github.com/pytorch/pytorch/pull/81761#issuecomment-1190249259

File: aten/src/ATen/native/LinearAlgebra.cpp [hunk 1]
```diff
@@ -2682,94 +2682,92 @@ Tensor& linalg_norm_out(const Tensor& X, c10::string_view ord, OptionalIntArrayR
...
-    auto dim_ = dim.vec();
-    maybe_wrap_dims(dim_, self.dim());
-    TORCH_CHECK(dim_[0] != dim_[1], "Expected dims to be different, got ", dim, " instead");
-    if (self.is_complex()) {
-      result = at::sqrt(at::sum(at::real(self.conj() * self), dim_, keepdim));
-    } else {
-      result = at::sqrt(at::sum((self * self), dim_, keepdim));
-    }
+  auto device = self.device();
+  if (self.layout() == Layout::Strided && (device == kCPU || device == kCUDA || device == kMeta)) {
```
Comment (ngimel) [inline aten/src/ATen/native/LinearAlgebra.cpp:2689]: nit - there are 6 (?) copies of this check with minor variations - time for a utility function?

**Review by ngimel — APPROVED** [review]

File: aten/src/ATen/native/LinearAlgebra.cpp (hunk 1 above)
Comment (Lezcano) [reply aten/src/ATen/native/LinearAlgebra.cpp:2689]: Fair enough

**Comment by mehtanirav** [comment]
This is causing some internal tests to fail with the following error.

```
terminate called after throwing an instance of 'c10::Error'
  what():  Could not find schema for aten::linalg_vector_norm.out
  
  
Exception raised from findSchemaOrThrow at xplat/caffe2/aten/src/ATen/core/dispatch/Dispatcher.cpp:131 (most recent call first):
(no backtrace available)
```

**Comment by kit1980** [comment]
I'll help to make sure next the merge will not break Meta-internal stuff.

**Comment by lezcano** [comment]
Thank you so much @kit1980

**Comment by lezcano** [comment]
Anything you need from my end just ping me

**Comment by kit1980** [comment]
@kit1980 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42354776).

**Comment by lezcano** [comment]
What's the state of this @kit1980?

**Comment by kit1980** [comment]
@lezcano Meta-internal mobile tests are failing. I'm fixing.

**Comment by lezcano** [comment]
Thank you!

**Comment by kit1980** [comment]
@lezcano Could you rebase?

**Comment by lezcano** [comment]
@kit1980 done!

**Comment by kit1980** [comment]
@kit1980 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42354776).

**Comment by kit1980** [comment]
Here is the XLA test that needs to be updated: https://github.com/pytorch/xla/blob/master/test/cpp/test_aten_xla_tensor.cpp#L1887

**Comment by kit1980** [comment]
Or we can just delete the test as did https://github.com/pytorch/xla/pull/3976
@lezcano what do you think?

**Comment by lezcano** [comment]
uh, no, we should fix that. Odd that that happened, ci was green...

**Comment by lezcano** [comment]
Right, this is an issue with some new symint stuff. I'll look into that.

**Comment by kit1980** [comment]
@lezcano don't reopen this PR, we didn't actually revert this.
Create a new PR to forward-fix the XLA test failure.

**Comment by lezcano** [comment]
Sure thing, will do that on Monday.

**Pull request closed without merging**